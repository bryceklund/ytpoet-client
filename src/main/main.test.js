import React from 'react'
import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer'
import { BrowserRouter } from 'react-router-dom'
import Main from './main'

describe(`Main`, () => {
    it(`renders without crashing`, () => {
        const div = document.createElement('div')
        ReactDOM.render(<BrowserRouter>
                        <Main />
                        </BrowserRouter>, div)
        ReactDOM.unmountComponentAtNode(div)
    })
    it(`matches its snapshot`, () => {
        const tree = renderer.create(<BrowserRouter>
                                    <Main />
                                    </BrowserRouter>)
                                    .toJSON()
        expect(tree).toMatchSnapshot()
    })
})