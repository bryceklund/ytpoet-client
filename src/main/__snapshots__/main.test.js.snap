// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main matches its snapshot 1`] = `
<div>
  <div
    className="input"
  >
    <form
      className="url-settings"
    >
      <fieldset>
        <legend>
          paste a video URL:
        </legend>
        <input
          className="url"
          onInput={[Function]}
          placeholder="https://www.youtube.com/watch?v=A_x1qvftnPs"
        />
        <a
          className="go hidden"
          href="/result"
          onClick={[Function]}
        >
          GO
        </a>
        <p
          className="error-message"
        >
          
        </p>
        <div
          className="setting-container"
        >
          <label
            htmlFor="poem-type"
          >
            poem type:
          </label>
          <select
            className="poem-type"
            name="poem-type"
            onChange={[Function]}
          >
            <option
              value="haiku"
            >
              haiku
            </option>
            <option
              value="sonnet"
            >
              sonnet
            </option>
            <option
              value="rondel"
            >
              rondel
            </option>
            <option
              value="indriso"
            >
              indriso
            </option>
            <option
              value="custom"
            >
              custom
            </option>
          </select>
        </div>
        <div
          className="settings hidden"
          id="settings"
        >
          <div
            className="setting-container"
          >
            <label
              htmlFor="syllables"
            >
              syllables per line:
            </label>
            <input
              className="syllables"
              name="syllables"
              onChange={[Function]}
              type="text"
            />
          </div>
          <div
            className="setting-container"
          >
            <label
              htmlFor="lines"
            >
              number of lines (1-20):
            </label>
            <input
              className="lines"
              name="lines"
              onChange={[Function]}
              type="text"
            />
          </div>
        </div>
        <div
          className="setting-container"
        >
          <label
            htmlFor="profanity"
          >
            block profanity?
          </label>
          <input
            name="profanity"
            onChange={[Function]}
            type="checkbox"
          />
        </div>
      </fieldset>
    </form>
  </div>
  <p
    className="sample-poems-label"
  >
    <a
      className="sample-poem"
      href="/poem/null"
    >
      random poem
    </a>
  </p>
  <script
    src="script.js"
  />
</div>
`;
